### DISCUSSION TITLE
Spring-starter and javaagent?

### DISCUSSION THREAD

**Discussion by Asafb26** [open-telemetry/opentelemetry-java-instrumentation#4610] [Q&A] [answered]
Hello.
We have some spring apps that handle rabbit, Kafka HTTP and GRPC traffic.
We want to include them in our OT distributed tracing across all of our other microservices, and I am quite lost with the options available for us here.
As far as I see, `opentelemetry-java-instrumentation` provides a `spring-starter` lib and a `javaagent` which add this support out of the box, but still I'm not sure what exactly is the difference between them. can someone please explain the differences?
Thanks

**Comment by trask** [comment, marked as answer]
hi @Asafb26! typically the easiest onboarding option is to use the javaagent, I'd recommend starting with that if it works for you

**Reply by Asafb26** [reply]
Thanks! Actually we already have a POC based on the javaagent, it is Indeed working but we have a limited control on it, we will tightly couple with baggage propagation and cannot afford a scenario like app running without javaagent (as far as I see I cannot throw an error if the javaagent crashed on initialization).
What exactly are the downside of spring-starter compared to the javaagent?

**Reply by trask** [reply]
> cannot afford a scenario like app running without javaagent (as far as I see I cannot throw an error if the javaagent crashed on initialization)

this seems reasonable to add a flag to the java agent for fast-failure, if you want to open an issue (or better a PR)

> What exactly are the downside of spring-starter compared to the javaagent?

there is less instrumentation available currently, and requires more manual work at this point. this situation should improve in the future
